test/common/common/hex_fuzz_test.cc

```diff
@@ -0,0 +1,24 @@
+#include "common/common/hex.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Fuzz {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+
+  FuzzedDataProvider fuzz_data(buf, len);
+  Envoy::Hex::uint64ToHex(fuzz_data.ConsumeIntegral<uint64_t>());
+  Envoy::Hex::uint32ToHex(fuzz_data.ConsumeIntegral<uint32_t>());
+
+  Envoy::Hex::encode(buf, len);
+  Envoy::Hex::decode(std::string(reinterpret_cast<const char*>(buf), len));
+
+  FuzzedDataProvider fuzz_codec_data(buf, len);
```

Comment:
Sorry for being nit-picky here, but if you consume the whole buffer for the round-trip, you don't need the provider and can just `FUZZ_ASSERT(std::vector<uint8_t>(buf, buf+len) == decoded`